AvailableProcessors: returns 0 in docker container

**Alluxio Version:**
2.3.0

**Describe the bug**

when I execute `alluxio ls`  in the container I got the following error.


```
bash-4.4$ alluxio fs ls /
Exception in thread "main" java.lang.ExceptionInInitializerError
        at io.grpc.netty.NettyChannelBuilder.<clinit>(NettyChannelBuilder.java:74)
        at alluxio.grpc.GrpcConnectionPool.createManagedChannel(GrpcConnectionPool.java:168)      
        at alluxio.grpc.GrpcConnectionPool.lambda$acquireConnection$0(GrpcConnectionPool.java:114)
        at java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1853)
        at alluxio.grpc.GrpcConnectionPool.acquireConnection(GrpcConnectionPool.java:86)
        at alluxio.grpc.GrpcChannelBuilder.build(GrpcChannelBuilder.java:121)
        at alluxio.util.ConfigurationUtils.loadConfiguration(ConfigurationUtils.java:495)
        at alluxio.ClientContext.loadConf(ClientContext.java:134)
        at alluxio.ClientContext.loadConfIfNotLoaded(ClientContext.java:155)
        at alluxio.AbstractClient.beforeConnect(AbstractClient.java:175)
        at alluxio.AbstractClient.connect(AbstractClient.java:224)
        at alluxio.client.file.BaseFileSystem.rpc(BaseFileSystem.java:518)
        at alluxio.client.file.BaseFileSystem.getStatus(BaseFileSystem.java:258)
        at alluxio.client.file.FileSystem.getStatus(FileSystem.java:322)
        at alluxio.cli.fs.command.LsCommand.ls(LsCommand.java:262)
        at alluxio.cli.fs.command.LsCommand.runPlainPath(LsCommand.java:320)
        at alluxio.cli.fs.command.AbstractFileSystemCommand.runWildCardCmd(AbstractFileSystemCommand.java:92)   
        at alluxio.cli.fs.command.LsCommand.run(LsCommand.java:330)
        at alluxio.cli.AbstractShell.run(AbstractShell.java:137)
        at alluxio.cli.fs.FileSystemShell.main(FileSystemShell.java:66)
        Suppressed: java.lang.NoClassDefFoundError: Could not initialize class io.grpc.netty.NettyChannelBuilder
                at alluxio.grpc.GrpcConnectionPool.createManagedChannel(GrpcConnectionPool.java:168)
                at alluxio.grpc.GrpcConnectionPool.lambda$acquireConnection$0(GrpcConnectionPool.java:114)      
                at java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1853)
                at alluxio.grpc.GrpcConnectionPool.acquireConnection(GrpcConnectionPool.java:86)
                at alluxio.grpc.GrpcChannelBuilder.build(GrpcChannelBuilder.java:121)
                at alluxio.util.ConfigurationUtils.loadConfiguration(ConfigurationUtils.java:495)
                at alluxio.ClientContext.loadConf(ClientContext.java:134)
                at alluxio.client.metrics.ClientMasterSync.loadConf(ClientMasterSync.java:123)
                at alluxio.client.metrics.ClientMasterSync.heartbeat(ClientMasterSync.java:75)
                at alluxio.client.metrics.MetricsHeartbeatContext.heartbeat(MetricsHeartbeatContext.java:95)
                at alluxio.client.metrics.MetricsHeartbeatContext.close(MetricsHeartbeatContext.java:125)
                at alluxio.client.metrics.MetricsHeartbeatContext.removeContext(MetricsHeartbeatContext.java:108)
                at alluxio.client.metrics.MetricsHeartbeatContext.removeHeartbeat(MetricsHeartbeatContext.java:198)
                at alluxio.client.file.FileSystemContext.closeContext(FileSystemContext.java:296)
                at alluxio.client.file.FileSystemContext.close(FileSystemContext.java:263)
                at com.google.common.io.Closer.close(Closer.java:214)
                at alluxio.cli.AbstractShell.close(AbstractShell.java:154)
                at alluxio.cli.fs.FileSystemShell.main(FileSystemShell.java:67)
Caused by: java.lang.IllegalArgumentException: availableProcessors: 0 (expected: > 0)
        at io.netty.util.internal.ObjectUtil.checkPositive(ObjectUtil.java:44)
        at io.netty.util.NettyRuntime$AvailableProcessorsHolder.setAvailableProcessors(NettyRuntime.java:44)
        at io.netty.util.NettyRuntime$AvailableProcessorsHolder.availableProcessors(NettyRuntime.java:70)
        at io.netty.util.NettyRuntime.availableProcessors(NettyRuntime.java:98)
        at io.grpc.netty.Utils$DefaultEventLoopGroupResource.<init>(Utils.java:394)
        at io.grpc.netty.Utils.<clinit>(Utils.java:84)
        ... 20 more
bash-4.4$
```

**To Reproduce**
execute `alluxioio fs ls` in the master container 

**Expected behavior**

no errors should be generated

**Urgency**

will be impacted 

**Additional context**

This error appeared in the log of the master and worker container and I fixed this error by adding following parameters to jvm

  - "-XX:ActiveProcessorCount=4"


ref: https://developers.redhat.com/blog/2017/04/04/openjdk-and-containers/

--- a/docs/cn/deploy/Running-Alluxio-On-Docker.md
+++ b/docs/cn/deploy/Running-Alluxio-On-Docker.md
@@ -316,3 +316,25 @@ $ docker run -e \
 ## 故障排除
 
 可以通过运行`docker logs $container_id`来访问Alluxio 服务器日志。通常，日志可以很好地指出什么地方出了问题。如果它们不足以诊断您的问题，则可以在[用户邮件列表](https://groups.google.com/forum/#!forum/alluxio-users).上获得帮助 。
+
+## FAQ
+
+### AvailableProcessors: returns 0 in docker container
+
+当在 alluxio master container 里执行 `alluxio fs ls /`，遇到如下报错时，
+
+```bash
+bash-4.4$ alluxio fs ls /
+Exception in thread "main" java.lang.ExceptionInInitializerError
+...
+Caused by: java.lang.IllegalArgumentException: availableProcessors: 0 (expected: > 0)
+        at io.netty.util.internal.ObjectUtil.checkPositive(ObjectUtil.java:44)
+        at io.netty.util.NettyRuntime$AvailableProcessorsHolder.setAvailableProcessors(NettyRuntime.java:44)
+        at io.netty.util.NettyRuntime$AvailableProcessorsHolder.availableProcessors(NettyRuntime.java:70)
+        at io.netty.util.NettyRuntime.availableProcessors(NettyRuntime.java:98)
+        at io.grpc.netty.Utils$DefaultEventLoopGroupResource.<init>(Utils.java:394)
+        at io.grpc.netty.Utils.<clinit>(Utils.java:84)
+        ... 20 more
+```
+
+可以通过增加 JVM 参数 `-XX:ActiveProcessorCount=4` 解决这个问题。

--- a/docs/en/deploy/Running-Alluxio-On-Docker.md
+++ b/docs/en/deploy/Running-Alluxio-On-Docker.md
@@ -485,3 +485,25 @@ Alluxio server logs can be accessed by running `docker logs $container_id`.
 Usually the logs will give a good indication of what is wrong. If they are not enough to diagnose
 your issue, you can get help on the
 [user mailing list](https://groups.google.com/forum/#!forum/alluxio-users).
+
+## FAQ
+
+### AvailableProcessors: returns 0 in docker container
+
+When you execute `alluxio ls` in the alluxio master container and got the following error.
+
+```bash
+bash-4.4$ alluxio fs ls /
+Exception in thread "main" java.lang.ExceptionInInitializerError
+...
+Caused by: java.lang.IllegalArgumentException: availableProcessors: 0 (expected: > 0)
+        at io.netty.util.internal.ObjectUtil.checkPositive(ObjectUtil.java:44)
+        at io.netty.util.NettyRuntime$AvailableProcessorsHolder.setAvailableProcessors(NettyRuntime.java:44)
+        at io.netty.util.NettyRuntime$AvailableProcessorsHolder.availableProcessors(NettyRuntime.java:70)
+        at io.netty.util.NettyRuntime.availableProcessors(NettyRuntime.java:98)
+        at io.grpc.netty.Utils$DefaultEventLoopGroupResource.<init>(Utils.java:394)
+        at io.grpc.netty.Utils.<clinit>(Utils.java:84)
+        ... 20 more
+```
+
+This error can be fixed by adding `-XX:ActiveProcessorCount=4` as jvm parameter.

